plugins/custom-codecs/src/main/java/org/opensearch/index/codec/customcodecs/CustomCodecService.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.codec.customcodecs;
+
+import org.apache.logging.log4j.Logger;
+import org.apache.lucene.codecs.Codec;
+import org.opensearch.common.collect.MapBuilder;
+import org.opensearch.index.IndexSettings;
+import org.opensearch.index.codec.CodecService;
+import org.opensearch.index.mapper.MapperService;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import static org.opensearch.index.engine.EngineConfig.INDEX_CODEC_COMPRESSION_LEVEL_SETTING;
+
+/**
+ * CustomCodecService provides ZSTD and ZSTD_NO_DICT compression codecs.
+ */
+public class CustomCodecService extends CodecService {
```

Comment:
Looking at the implementation, it is not generic enough to support any custom codec. Rather than calling it ```Custom```, can it be changed to ```ZStd``` to align with the implementation?